### PR TITLE
[QoL] Show move description via info button

### LINKED ISSUE #3777: (Qol) To see move descriptions while battle in the fight tab
### Is your feature request related to a problem? Please describe.

No

### Describe the Feature

At the fight tab, has the move list, maybe when user click a button while choosing one of the move, the move descriptions show up

Screenshot_20240826_184452_Chrome

### Additional context

This for easier to know what the move are while fighting, so the user don't need to go to pokemon, the summary then can see the move descriptions

### PR DESCRIPTION
**Pull request by gruxor**
## What are the changes the user will see?
While holding the info toggle button (C/"Back"/Select) with the FIGHT menu active, the description of the currently selected move's effect will be displayed. This is only active if the user has the existing "Move Info" option enabled.

## Why am I making these changes?
Large quality of life improvement, especially for users who don't have encyclopedic knowledge of move effects.
This also causes parity with behavior in the modifier select screen by utilizing the same functionality that TMs use.

Also closes #3777 by coincidence, found via search - hopefully didn't miss any other related issues/PRs.

## What are the changes from a developer perspective?
This uses the somewhat under-utilized `addInfoToggle` system in the Battle Scene (previously only used by the TM overlay) to avoid throwing this functionality into the otherwise somewhat hard-coded stats overlay system. It mainly uses the same `MoveInfoOverlay` class that the TM overlay uses, so it remains fairly simple. The necessary modifications were added via optional flags passed via `MoveInfoOverlaySettings`. On the `FightUiHandler` side, it was necessary to implement the `InfoToggle` interface in order to hide the move names and cursor while the description is active.

One minor side-effect of the changes is that the TM description/info overlay received a size change (2 pixels on each axis, give or take, iirc) due to the window not being identical in size to the `FightUiHandler` window, but it appears the difference was arbitrary and parity between the two is likely a slight improvement. Difference not even noticable unless you're comparing side by side. 

Additionally, since I implemented the tween fade in order to match the style of the stats overlays, the TM overlay will also fade now which looks a bit nicer.

## How to test the changes?
Open any battle and hold the info button (C default). No automated tests created (or needed, probably).
`{name: "TM_COMMON"}` can be added to `ITEM_REWARD_OVERRIDE` to test any potential issues with the existing TM functionality, but the changes there are negligible (if not unnoticable).

## Other notes
First PR on this repo and first real experience tinkering with TS and Phaser (I'm typically a C and C# guy + Godot/Unity with a bit of other stuff sprinkled in) so hopefully I didn't commit any deadly sins here. If I did, feel free to crucify me and I'll try to patch things up from there.

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by gruxor** [comment]
Testing some long localized descriptions as requested by @frutescens on Discord:
image
image
image
image

The only language that appears to have issues is Japanese:
image
This is an issue inherited from a current issue that also exists in the TM description and isn't introduced by this PR. When it is addressed though, it would fix it for both the original TM menu and this battle implementation as well since they utilize the same window.

**Review by KimJeongSun — COMMENTED** [review]
I like this enhancement!

File: src/ui/move-info-overlay.ts [hunk 1]
```diff
@@ -16,11 +16,13 @@ export interface MoveInfoOverlaySettings {
     x?: number;
     y?: number;
     width?: number; // default is always half the screen, regardless of scale
+    hideEffectBox?: boolean; // whether or not the small secondary box should be shown
+    hideBg?: boolean; // whether to hide the window background
```
Comment (DayKev) [inline src/ui/move-info-overlay.ts:20]:
```suggestion
    /** Default is always half the screen, regardless of scale */
    width?: number;
    /* Determines whether to display the small secondary box */
    hideEffectBox?: boolean;
    hideBg?: boolean;
```

**Comment by LagendaBergerak** [comment]
Thanks man, this will help a lot

**Comment by CodeTappert** [comment]
Works fine. Even scrolls if the text is larger then it could fit in the textbox

**Review by CodeTappert — APPROVED** [review]

**Review by Opaque02 — APPROVED** [review]

**Review by KimJeongSun — APPROVED** [review]
LGTM

**Review by ben-lear — APPROVED** [review]

**Pull request merged**